collectors/EsxiOverallStateCollector.py

```diff
@@ -0,0 +1,65 @@
+from os import getenv
+from queue import Queue
+from threading import Thread
+from prometheus_client.core import GaugeMetricFamily
+from BaseCollector import BaseCollector
+
+import logging
+import modules.Configuration as config
+
+# init logging
+logger = logging.getLogger('esxi-exporter')
+
+
+class EsxiOverallStateCollector(BaseCollector):
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    def worker(q: Queue, output: dict):
+        while not q.empty():
+            host = q.get()
+            output[host.name] = host.overallStatus
+
+    def describe(self):
+        yield GaugeMetricFamily('esxi_overall_state',
+                                'overall health status of esxi-host collected from vCenter')
+
+    def collect(self):
+
+        gauge_metric = GaugeMetricFamily('esxi_overall_state',
+                                         '0=unknown 1=red, 2=orange, 3=green',
+                                         labels=['vcenter', 'hostsystem'])
+
+        # get all hosts from vcenter
+        vc_hosts = self.get_vcenter_hosts()
+
+        # prepare multithreading
+        threads = []
+        results = {}
+        q = Queue()
+        [q.put(host) for host in vc_hosts]
+
+        for i in range(config.overallstate_threads):
+            t = Thread(target=EsxiOverallStateCollector.worker,
+                       args=(q, results))
+            threads.append(t)
+            t.start()
+
+        [t.join() for t in threads]
+
+        # parse results
+        for host, value in results.items():
+            if value == 'green':
+                state = 3
+            elif value == 'yellow':
+                state = 2
+            elif value == 'red':
+                state = 1
+            else:
+                state = 0
+            gauge_metric.add_metric(
+                labels=[getenv('VCENTER_URL'), host, 'ssh_connection'],
```

Comment:
All parameter parsing need to go to `exporter.py`. Don't do any environment var handling in the modules please.

Reply:
Lol I once had this all in a class.